### ISSUE TITLE
GIoULoss doesn't go below 1

### ISSUE THREAD

**Issue by nimish1512** [tensorflow/addons#982]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04 LTS
- TensorFlow version and how it was installed (source or binary): 2.1.0 (binary)
- TensorFlow-Addons version and how it was installed (source or binary):0.7.1 (binary)
- Python version: 3.6.9
- Is GPU used? (yes/no): yes

**Describe the bug**
I am using the GIoULoss to train a small CNN using keras which predicts the (x,y) of a bounding box. The training starts with loss at 1.04 and stops decreasing after 1.0. I tried changing the architecture and epochs but it gets stuck at val_loss=1.0

**Code to reproduce the issue**

```
def PreprocessLabels(row,col,rad):
    # Convert (row,col) & radius to (x1,y1) & (x2,y2)
    # This helps us to shape the problem as bounding box prediction problem
    # where we can use GIoU as our loss function.
    x1 = int(row)-int(rad)
    y1 = int(col)-int(rad)
    x2 = int(row)+int(rad)
    y2 = int(col)+int(rad)
    return [y1,x1,y2,x2]

    def DataGenerator(FileList,mode='train'):
        DataInput, DataLabels = [],[]
            for img_name in FileList:
                img = cv2.imread(img_name,2)
                row,col,rad = img_name.split("_")[2:]
                rad = rad.split(".")[0]
                DataInput.append(img)
                DataLabels.append(PreprocessLabels(row,col,rad))
        return np.expand_dims(np.array(DataInput),axis=3),np.array(DataLabels)


def Model():
    model = Sequential()
    model.add(Conv2D(64,kernel_size=3,activation='relu',input_shape=(200,200,1)))
    model.add(Conv2D(64,kernel_size=5,activation='relu'))
    model.add(Conv2D(32,kernel_size=3,activation='relu'))
    model.add(Flatten())
    model.add(Dense(4,activation='linear',use_bias=False,kernel_initializer="he_normal"))
    return model

FileList = glob("images/*.png")
x,y = DataGenerator(FileList)

model = Model()
model.compile(optimizer='adam',loss=GIoULoss())
model.fit(x,y,batch_size=64,epochs=10,verbose=1,validation_split=0.2,shuffle=True,)
```

**Comment by seanpmorgan** [comment]
@nimish1512 Thanks for raising the issue. Unfortunately we can't ask maintainers to troubleshoot training code as there is no expected result (especially given the unknown dataset your using). 

Please create a minimal reproducible example (similar to a test case) that can be verified by a maintainer. For example if you think GIoULoss doesn't go below 1, then create an example input that should and show how its broken.

Apologies for the less-than-helpful answer but we have to keep maintainability a priority. Having maintainers debug entire ML pipelines is very difficult and time consuming.

**Comment by betamoo** [comment]
I am having the same problem as well, the loss reaches 1 and stays there!

**Comment by FabianMoenkeberg** [comment]
Same problem here...

**Comment by JoelMaldonadoRuiz** [comment]
Same problem as well.

**Comment by JoelMaldonadoRuiz** [comment]
Seems like changing the batch size fixed it for me

**Comment by JoelMaldonadoRuiz** [comment]
Seems like normalizing the data fixed it for me

**Comment by JoelMaldonadoRuiz** [comment]
Seems like adding normalization fixed it for me

**Comment by akshayasthana1** [comment]
Hi All. Can someone please confirm if this issue has been fixed? The last comment is too cryptic. Thanks in advance!
